pkg/kubectl/service.go

```diff
@@ -183,8 +183,16 @@ func generate(genericParams map[string]interface{}) (runtime.Object, error) {
 	if len(params["type"]) != 0 {
 		service.Spec.Type = api.ServiceType(params["type"])
 	}
-	if service.Spec.Type == api.ServiceTypeLoadBalancer {
+	switch service.Spec.Type {
+	case api.ServiceTypeNodePort:
+		for i, port := range service.Spec.Ports {
+			service.Spec.Ports[i].NodePort = port.Port
+		}
+	case api.ServiceTypeLoadBalancer:
 		service.Spec.LoadBalancerIP = params["load-balancer-ip"]
+	case "Headless":
```

Comment:
@thockin there is a --load-balancer-ip for assigning IPs to LoadBalancers and the command has already a bunch of flags. It feels that there should be subcommands here for the different types of services. WDYT?